clients/rospy/src/rospy/timer.py

```diff
@@ -204,7 +207,14 @@ def shutdown(self):
         Stop firing callbacks.
         """
         self._shutdown = True
-        
+        self._period = self.__period
+
+    def startup(self):
```

Comment:
Also, calling `start()` is not necessary. In this case you have a Timer object whose `_shutdown` property is True (`run` is not executing its main logic block), yet the thread still exists. Calling `startup()` flips the bit so that the `run` method begins paying attention to its main logic block again.